### DISCUSSION TITLE
Libre 2 EU direct connection, intermittent BT status on Samsung A33 Android 14

### DISCUSSION THREAD

**Discussion by Paul-0000** [NightscoutFoundation/xDrip#3755] [Q&A]
Hi all, 
I cannot achieve a permanent direct Libre2 EU Bluetooth connection between the sensor & xDrip+ at my daily used Samsung Galaxy A33 while this works perfectly on my old Samsung Galaxy J5 
[see details at end].
  
For the J5 the xDrip+ system status page shows all the time Bluetooth Device: connected. 
xDrip+ event log and OOP2 show last bg readings for every minute, all works fine.

For the A33 the xDrip+ the system status page shows the Bluetooth device connects only for about a second followed by disconnected state, this in a repetitive fashion.  
The Phone Service State shows: Enabled BlueCon for (tens of) minutes and then changes to reconnect. In about one hour no connection can be achieved successfully anymore. For the best case the xDrip event log and OOP2 show last bg readings for every two minutes. As far as I can see there are no obvious failures  between these readings in the event log.

I tested various advanced BT settings as describeb by xDrip+, disabled optimization for battery, BT, OOP2 and xDrip+, set Location Service always on, etc.

Can it be this intermittent connection behavior for my A33 is due to the current Android 14 version and I cannot solve this in settings?
This as I noticed a xDrip message during installation showing ‘’designed for older Android version’’.
Furthermore, Android 14 seems to have several Bluetooth issues.
Any thoughts here?

Thank you for all the great work on xDrip+! 

Location NL, Sensor Libre2-EU direct, no bridge
Samsung Galaxy A33 5G Android 14 One UI 6.1
Samsung Galaxy J5 Android 9 One UI1.1 
xDrip+ version d14910c-2024.10.29
OOP2

**Comment by PhilTU** [comment]
https://xdrip.readthedocs.io/en/latest/troubleshoot/libre2direct-qr-samsung/ might be worth a look. There are some xDrip+ settings you can try different combinations.

**Reply by Paul-0000** [reply]
I tried that earlier but unfortunately no joy, thank you for suggesting

**Reply by parapenT1sta** [reply]
I am using a Samsung Z Fold 4 and the options I use are different from the site shared above. If you want I can share with you.

**Reply by Paul-0000** [reply]
Yes, sharing would be nice. 
Furthermore, for what type of sensor and Android version do you apply those settings?

**Reply by parapenT1sta** [reply]
I have been using these settings since I got Samsung phones, for other Android Phones (Google Pixel) I was using other settings.

I shared my settings before on this topic: https://github.com/NightscoutFoundation/xDrip/discussions/3481#discussioncomment-9594730

**Reply by Paul-0000** [reply]
As to the suggestions in your thread, I experimented with variations in active/deactive state of "Agressive Service Restart, "G5 BT Watchdog, and ''Companion BT". Unfortunately, it did not help to solve the repetitive quick disconnect behavior.

**Reply by parapenT1sta** [reply]
In fact these were the settings that really helped me:

Close GATT = Unchecked
Use Scanning = Unchecked
Trust AutoConnect = Checked
Use Background Scans = Checked
Companion BT = Checked
Bluetooth Wakelock = Unchecked
Constantly Reset BT = Unchecked

![1](https://github.com/user-attachments/assets/23a1f4f6-0af7-4a89-ba4e-8525ff2ec603)

**Reply by Paul-0000** [reply]
Your settings were included in my testing as were navid200's
https://navid200.github.io/xDrip/docs/BatteryOpt.html
https://navid200.github.io/xDrip/docs/BluetoothBatteryOpt.html
But stil no change, BT status keeps showing 'connected' only for a second or so, and the loop starts again....
